observers/slack_resource.cpp

```diff
@@ -0,0 +1,109 @@
+#include <memory>
+#include <vector>
+
+#include "glog/logging.h"
+
+#include "mesos/mesos.hpp"
+
+#include "serenity/executor_set.hpp"
+#include "serenity/metrics_helpers.hpp"
+
+#include "observers/slack_resource.hpp"
+
+namespace mesos {
+namespace serenity {
+
+Try<Nothing> SlackResourceObserver::consume(const ResourceUsage& usage) {
+  std::unique_ptr<ExecutorSet> newSamples(new ExecutorSet());
+  std::vector<Resource> slackResources;
+
+  for (const auto& executor : usage.executors()) {
+    if (executor.has_statistics() && executor.has_executor_info()) {
+      newSamples->insert(executor);
+
+      auto previousSample = this->previousSamples->find(executor);
+      if (previousSample != this->previousSamples->end()) {
+        Result<Resource> slackResource = CalculateSlack(
+            (*previousSample), executor);
+        if (slackResource.isSome()) {
+          slackResources.push_back(slackResource.get());
```

Comment:
Just let me undestand that. Why are you making special vector for cpu Resources (if you want a list of resources just use ready Resources class btw) and then you are just reducing it in a seperate loop and create another seperate revocable resource... Why don't you just make a double_t or one revocable Resource which will sum slack cpu in run time, here? You can then get rid of CombineSlack method at all. ((: Maybe there is a reason for that which i'm not aware..